app/src/main/java/com/sun/mygif/data/source/remote/response/GetResponseAsync.kt

```diff
@@ -0,0 +1,35 @@
+package com.sun.mygif.data.source.remote.response
+
+import android.os.AsyncTask
+import com.sun.mygif.data.model.DataRequest
+import com.sun.mygif.data.source.local.base.OnDataLoadedCallback
+import org.json.JSONException
+import java.io.IOException
+import java.lang.Exception
+
+class GetResponseAsync<T>(
+    private val responseHandler: DataResponseHandler<T>,
+    private val callback: OnDataLoadedCallback<T>
+) : AsyncTask<DataRequest, Void, Exception?>() {
+
+    private var result: T? = null
+
+    override fun doInBackground(vararg params: DataRequest): Exception? =
```

Comment:
hàm getData a nghĩ đầu vào là 1 url có vẻ hợp lý hơn
```suggestion
    override fun doInBackground(vararg params: String): Exception? =
```